File: src/types/knowledge-domain/analysis.ts
import type {
  GenerateKnowledgeCodeDocumentsResult,
  KnowledgeCodeAnalysisResult,
  KnowledgeCodeSnapshot,
  KnowledgeCodeSource,
  UpsertKnowledgeCodeSourceInput,
} from "@/types/knowledge";

export type {
  GenerateKnowledgeCodeDocumentsResult,
  KnowledgeCodeAnalysisResult,
  KnowledgeCodeSnapshot,
  KnowledgeCodeSource,
  UpsertKnowledgeCodeSourceInput,
};

/** 仅查询该项目的代码源；项目范围在 Rust IPC 边界再次校验。 */
export interface ListKnowledgeAnalysisCodeSnapshotsInput {
  projectId: number;
  sourceId?: number | null;
}

/**
 * 新工作台以 projectVersionId 表述版本范围，领域 API 在兼容层映射为旧快照接口的
 * releaseId；不传版本时只能形成未绑定发布版本的快照，不能当作发布事实。
 */
export interface CaptureKnowledgeAnalysisGitSnapshotInput {
  projectId: number;
  sourceId: number;
  gitRef: string;
  projectVersionId?: number | null;
}

export interface AnalyzeKnowledgeAnalysisSnapshotInput {
  projectId: number;
  snapshotId: number;
}

export interface GenerateKnowledgeAnalysisDocumentsInput {
  projectId: number;
  snapshotId: number;
}

export interface CreateKnowledgeAnalysisDraftInput {
  projectId: number;
  projectVersionId: number;
  snapshotIds: number[];
  /** 留空时由后端选择当前已配置的聊天 Provider。 */
  providerKey?: string | null;
  templateKey?: string | null;
}

export interface KnowledgeAnalysisDraft {
  id: number;
  analysisRunId: number;
  projectId: number;
  projectVersionId: number;
  snapshotIds: number[];
  providerKey: string;
  model: string;
  templateKey: string;
  content: string;
  claimRefs: string[];
  status: string;
  confirmedDocumentVersionId?: number | null;
}

export interface ConfirmKnowledgeAnalysisDraftInput {
  draftId: number;
  title: string;
  content: string;
  versionLabel: string;
  authorLabel?: string | null;
}

export interface ConfirmKnowledgeAnalysisDraftResult {
  draft: KnowledgeAnalysisDraft;
  document: {
    documentId: number;
    documentVersionId: number;
    parentVersionId?: number | null;
    contentHash: string;
    indexJobId: number;
    indexJobStatus: string;
  };
}

File: src-tauri/src/models/knowledge_domain/analysis.rs
use serde::{Deserialize, Serialize};

/// AI 分析草稿只能引用已冻结的快照和证据；确认入库由独立提交操作完成。
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateKnowledgeAnalysisDraftInput {
    pub project_id: i64,
    pub project_version_id: i64,
    pub snapshot_ids: Vec<i64>,
    #[serde(default)]
    pub provider_key: Option<String>,
    #[serde(default)]
    pub template_key: Option<String>,
}

/// AI 生成的项目分析草稿。它与正式知识文档版本分离，用户编辑并确认前不会进入搜索、
/// 图谱或问答索引。
#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct KnowledgeAnalysisDraft {
    pub id: i64,
    pub analysis_run_id: i64,
    pub project_id: i64,
    pub project_version_id: i64,
    pub snapshot_ids: Vec<i64>,
    pub provider_key: String,
    pub model: String,
    pub template_key: String,
    pub content: String,
    /// 每一个引用都指向本次固定代码快照中的一个文件，而不是由模型自由编造路径。
    pub claim_refs: Vec<String>,
    pub status: String,
    pub confirmed_document_version_id: Option<i64>,
}

/// 用户确认入库时可以编辑标题与正文；后端会创建新的不可变知识文档版本，而非覆盖 AI
/// 草稿或任意既有文档。
#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ConfirmKnowledgeAnalysisDraftInput {
    pub draft_id: i64,
    pub title: String,
    pub content: String,
    pub version_label: String,
    #[serde(default)]
    pub author_label: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ConfirmKnowledgeAnalysisDraftResult {
    pub draft: KnowledgeAnalysisDraft,
    pub document: crate::models::knowledge_domain::documents::KnowledgeDocumentCommitResult,
}
